integration-tests/redis_test.go

```diff
@@ -130,6 +163,32 @@ var _ = Describe("Redis", Label("Redis"), func() {
 			Entry("labels", "labels", "a-valid-list-of-labels"),
 		)
 
+		It("should prevent passing the deprecated property `cache_size` as user input", func() {
+			_, err := broker.Provision(redisServiceName, "custom-sample", map[string]any{"cache_size": 2})
+			Expect(err).To(MatchError(ContainSubstring("additional properties are not allowed: cache_size")))
+		})
```

Comment:
thoughts(non-blocking): I see the value of the test but, I am not sure if the text is accurate because we cannot update `cache_size` due to it being a plan property rather than an additional property or deprecated

Reply:
This condition was included for acceptance:
> keep cache_size as a plan property, so that users can't pass it in by parameter. Remove validation.

So I thought it would be valuable to represent as a test to ensure that future us don't move the property to be a user input (at least not without breaking this test).

Reply:
Maybe the intent of the test would be clearer if I remove the "MatchError(ContainSubstring" condition, and just leave that the provision should fail.

Reply:
The property `cache_size` hasn't been ever an allowed user input but this error was hidden because it was a mandatory property plan and the "should prevent modifying `plan defined properties`" error was thrown before this one.